core/offchain/impl/http_request.hpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_OFFCHAIN_HTTPREQUEST
+#define KAGOME_OFFCHAIN_HTTPREQUEST
+
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/context.hpp>
+#include <boost/asio/ssl/stream.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/core/tcp_stream.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/ssl/ssl_stream.hpp>
+
+#include "log/logger.hpp"
+#include "offchain/impl/uri.hpp"
+#include "offchain/types.hpp"
+
+namespace kagome::offchain {
+
+  class HttpRequest final : public std::enable_shared_from_this<HttpRequest> {
```

Comment:
I think it might be better to split the logic of constructing a request from the logic of sending it. It is quite non-obvious which methods relate to the construction and which send the data over a network.

Reply:
Process of connecting starts as soon as minimal required data has provided. It is usually happening while init phase (host and port is enough for that). HTTP-parts of request constructing is making parallel and independently. All implemented in according with the spec.